Source/PlasticSourceControl/Private/PlasticSourceControlMenu.cpp

```diff
@@ -529,6 +536,41 @@ void FPlasticSourceControlMenu::VisitLockRulesURLClicked(const FString InOrganiz
 	FPlatformProcess::LaunchURL(*OrganizationLockRulesURL, NULL, NULL);
 }
 
+void FPlasticSourceControlMenu::OpenDeskoptApp() const
+{
+#if PLATFORM_WINDOWS
+	// On Windows, use the registry to find the install location
+	FString InstallLocation = TEXT("C:/Program Files/PlasticSCM5");
+	if (FWindowsPlatformMisc::QueryRegKey(HKEY_LOCAL_MACHINE, TEXT("SOFTWARE\\Unity Software Inc.\\Unity DevOps Version Control"), TEXT("Location"), InstallLocation))
+	{
+		FPaths::NormalizeDirectoryName(InstallLocation);
+	}
+
+	const TCHAR* PlasticExe = TEXT("client/plastic.exe");
+	const TCHAR* GluonExe = TEXT("client/gluon.exe");
+	const FString DeskoptAppPath = FPaths::Combine(InstallLocation, FPlasticSourceControlModule::Get().GetProvider().IsPartialWorkspace() ? GluonExe : PlasticExe);
+#elif PLATFORM_MAC
+	const TCHAR* PlasticExe	= "/Applications/PlasticSCM.app/Contents/MacOS/macplasticx";
+	const TCHAR* GluonExe	= "/Applications/Gluon.app/Contents/MacOS/macgluonx";
+	const TCHAR* DeskoptAppPath = FPlasticSourceControlModule::Get().GetProvider().IsPartialWorkspace() ? GluonExe : PlasticExe);
+#elif PLATFORM_LINUX
+	const TCHAR* PlasticExe	= "/usr/bin/plasticgui ";
+	const TCHAR* GluonExe	= "/usr/bin/gluon";
+	const TCHAR* DeskoptAppPath = FPlasticSourceControlModule::Get().GetProvider().IsPartialWorkspace() ? GluonExe : PlasticExe);
+#endif
```

Comment:
For better readability, I'd recommend to extract this well-packed logic to a separate method or utility class, returning the current `DeskoptAppPath`.

Reply:
I moved this OS dependant logic into a new PlasticSourceControlUtils::FindDesktopApplicationPath()